newrelic-agent/src/main/java/com/newrelic/agent/service/logging/LogSenderServiceImpl.java

```diff
@@ -176,6 +177,22 @@ private void removeHarvestables() {
         }
     }
 
+    private static boolean isValidMapEntry(Map.Entry<String, ?> attribute) {
+        // key or value is null, skip it with a log message and iterate to next entry in attributes.entrySet()
+        if (attribute == null || attribute.getKey() == null || attribute.getValue() == null) {
+            Agent.LOG.log(Level.WARNING, "Log event with invalid attributes key or value of null was reported for a transaction but ignored."
+                    + " Each key should be a String and each value should be a String, Number, or Boolean.");
+            return false;
+        }
+        return true;
+    }
+
+    private static Map<String, ?> filterNulls(Map<String, ?> attributes) {
+        return attributes.entrySet().stream()
+                .filter(LogSenderServiceImpl::isValidMapEntry)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
     /**
```

Comment:
We don't need all of this extra filtering logic, it is already done in the `createValidatedEvent` method. In actual usage it will not be possible to ever have an empty attribute map because at a minimum it will always contain some agent metadata attributes. 

I don't think we need to add all of this additional code complexity and processing overhead just for a case that can only exist in a contrived unit test scenario. Using the streaming API is an extremely heavy operation when you're operating on a small map and this code needs to be as efficient as possible since it is happening for every single message that gets logged. Can we please revert back to what we had previously?

Reply:
I left it in a single commit in case it was not desired so it could be easily removed.

Reply:
Done.